src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorExtensions.cs

```diff
@@ -44,6 +44,46 @@ public static Tensor<T> Resize<T>(Tensor<T> input, ReadOnlySpan<nint> shape)
         #endregion
 
         #region Broadcast
+        /// <summary>
+        /// Broadcast the data from <paramref name="left"/> to the smallest broadcastable shape compatible with <paramref name="right"/>. Creates a new <see cref="Tensor{T}"/> and allocates new memory.
+        /// If the shapes are not compatible, <see cref="bool"/> is returned.
+        /// </summary>
+        /// <param name="left">Input <see cref="Tensor{T}"/>.</param>
+        /// <param name="right">Other <see cref="Tensor{T}"/> to make shapes broadcastable.</param>
+        /// <param name="destination">Destination <see cref="Tensor{T}"/>.</param>
+        /// <returns></returns>
+        public static bool TryBroadcastTo<T>(Tensor<T> left, Tensor<T> right, out Tensor<T> destination)
+            where T : IEquatable<T>, IEqualityOperators<T, T, bool>
+        {
+            try
+            {
+                nint[] newSize = TensorHelpers.GetSmallestBroadcastableSize(left.Lengths, right.Lengths);
+
+                Tensor<T> intermediate = BroadcastTo(left, newSize);
+                destination = Tensor.Create(intermediate.ToArray(), intermediate.Lengths);
+                return true;
+            }
+            catch
```

Comment:
There's no way to implement this without using exceptions for control flow? This largely defeats the intended purpose of the Try pattern, which is to avoid the expense of exceptions for the condition represented by the Try.